network/lwip2transport/udp.go

```diff
@@ -55,13 +55,15 @@ func (h *udpHandler) ReceiveTo(tunConn lwip.UDPConn, data []byte, destAddr *net.
 
 	h.mu.Lock()
 	reqSender, ok := h.senders[laddr]
+	h.mu.Unlock()
 	if !ok {
 		if reqSender, err = h.newSession(tunConn); err != nil {
```

Comment:
I would suggest adding a `h.mu.Unlock()` before returning. The goal is to protect both `h.senders[laddr]` read and `h.senders[laddr] = reqSender` write.

Reply:
Unfortunately, this is not possible since `newSession` will do `h.mu.Lock()` as well, so if it's not unlocked by then, it will always deadlock. As I understand, this execution path will always cause deadlock in the current code base.

I understand there may be an issue regarding access of `h.senders[laddr]`, but fixing it requires uprooting way more code that I understand at the moment. So I suggest to fix the hard deadlock issue now with minimal change - if there are any issues with UDP sessions access synchronization in the future, they may be taken care of separately. After all, this piece of code was essentially a dead end so there may be more bugs there.

Reply:
Perhaps I didn't understand you well and perhaps I didn't explain the issue well, so it's not possible to always do all `h.senders[laddr]` access inside the same mutex as they are interleaved with `newSession()` sometimes which requires mutex to be unlocked first.

Reply:
Thanks for the explanation. It's acceptable to keep your implementation. Would you mind adding a TODO comment here to mention that we need a `newSessionUnlock` for better synchronization?